File: android/app/src/main/java/com/igbocalendar/ui/screens/CalendarScreen.kt
package com.igbocalendar.ui.screens

import androidx.compose.animation.*
import androidx.compose.animation.core.*
import androidx.compose.foundation.*
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.rounded.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.igbocalendar.data.CalendarView
import com.igbocalendar.data.IgboCalendarEngine
import com.igbocalendar.ui.components.DateSearchBottomSheet
import com.igbocalendar.ui.components.GregorianCalendarGrid
import com.igbocalendar.ui.components.IgboDateCard
import com.igbocalendar.ui.components.IgboMonthGrid
import com.igbocalendar.ui.theme.igboColors
import com.igbocalendar.viewmodel.CalendarUiState
import java.time.LocalDate
import java.time.format.DateTimeFormatter

@Composable
fun CalendarScreen(
    state: CalendarUiState,
    onDateSelected: (LocalDate) -> Unit,
    onGregorianPrevMonth: () -> Unit,
    onGregorianNextMonth: () -> Unit,
    onIgboPrevMonth: () -> Unit,
    onIgboNextMonth: () -> Unit,
    onViewChange: (CalendarView) -> Unit,
    onToggleTheme: () -> Unit,
    onGoToday: () -> Unit,
    modifier: Modifier = Modifier
) {
    val colors = igboColors
    val scrollState = rememberScrollState()
    var showDateSearch by remember { mutableStateOf(false) }

    Column(
        modifier = modifier
            .fillMaxSize()
            .background(colors.background)
            .verticalScroll(scrollState)
    ) {
        // ── App Header ────────────────────────────────────────────────────────
        AppHeader(
            onToggleTheme = onToggleTheme,
            onGoToday = onGoToday,
            onGoToDate = { showDateSearch = true }
        )

        // ── Calendar View Toggle ──────────────────────────────────────────────
        ViewToggleBar(
            currentView = state.calendarView,
            onViewChange = onViewChange
        )

        Spacer(Modifier.height(8.dp))

        // ── Calendar Panel ────────────────────────────────────────────────────
        AnimatedContent(
            targetState = state.calendarView,
            transitionSpec = {
                fadeIn(tween(200)) togetherWith fadeOut(tween(200))
            },
            label = "calendar_view"
        ) { view ->
            when (view) {
                CalendarView.GREGORIAN -> GregorianPanel(
                    state = state,
                    onDateSelected = onDateSelected,
                    onPrevMonth = onGregorianPrevMonth,
                    onNextMonth = onGregorianNextMonth
                )
                CalendarView.IGBO_ERA -> IgboEraPanel(
                    state = state,
                    onDateSelected = onDateSelected,
                    onPrevMonth = onIgboPrevMonth,
                    onNextMonth = onIgboNextMonth
                )
            }
        }

        Spacer(Modifier.height(16.dp))

        // ── Selected Date Card ────────────────────────────────────────────────
        IgboDateCard(
            selectedDate = state.selectedDate,
            igboDate = state.igboDate,
            modifier = Modifier.padding(horizontal = 16.dp)
        )

        // ── Izu Week Info Banner ──────────────────────────────────────────────
        if (!state.igboDate.isUboshiOmumu) {
            Spacer(Modifier.height(12.dp))
            IzuInfoBanner(state = state)
        }

        Spacer(Modifier.height(24.dp))
    }

    if (showDateSearch) {
        DateSearchBottomSheet(
            currentDate = state.selectedDate,
            currentIgboDate = state.igboDate,
            onNavigate = onDateSelected,
            onDismiss = { showDateSearch = false }
        )
    }
}

@Composable
private fun AppHeader(
    onToggleTheme: () -> Unit,
    onGoToday: () -> Unit,
    onGoToDate: () -> Unit
) {
    val colors = igboColors
    val isAmber = colors.appTheme == com.igbocalendar.data.AppTheme.AMBER

    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = 20.dp, vertical = 16.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.SpaceBetween
    ) {
        Column {
            Text(
                text = "Igbo Calendar",
                fontSize = 22.sp,
                fontWeight = FontWeight.Black,
                color = colors.primary
            )
            Text(
                text = "Ọha Igbo",
                fontSize = 12.sp,
                color = colors.onSurfaceMuted,
                fontWeight = FontWeight.Medium
            )
        }

        Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
            // Go to date button
            Box(
                modifier = Modifier
                    .size(36.dp)
                    .clip(androidx.compose.foundation.shape.CircleShape)
                    .background(colors.surfaceVariant)
                    .clickable(onClick = onGoToDate),
                contentAlignment = Alignment.Center
            ) {
                Icon(
                    Icons.Rounded.Search,
                    contentDescription = "Go to date",
                    tint = colors.primary,
                    modifier = Modifier.size(18.dp)
                )
            }

            // Today button
            Box(
                modifier = Modifier
                    .clip(RoundedCornerShape(20.dp))
                    .background(colors.primaryContainer)
                    .clickable(onClick = onGoToday)
                    .padding(horizontal = 14.dp, vertical = 7.dp)
            ) {
                Text(
                    text = "Today",
                    fontSize = 13.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = colors.primary
                )
            }

            // Theme toggle
            Box(
                modifier = Modifier
                    .size(36.dp)
                    .clip(androidx.compose.foundation.shape.CircleShape)
                    .background(colors.surfaceVariant)
                    .clickable(onClick = onToggleTheme),
                contentAlignment = Alignment.Center
            ) {
                Text(
                    text = if (isAmber) "🌙" else "🌕",
                    fontSize = 16.sp
                )
            }
        }
    }
}

@Composable
private fun ViewToggleBar(
    currentView: CalendarView,
    onViewChange: (CalendarView) -> Unit
) {
    val colors = igboColors

    Row(
        modifier = Modifier
            .padding(horizontal = 20.dp)
            .clip(RoundedCornerShape(12.dp))
            .background(colors.surfaceVariant)
            .padding(4.dp),
        horizontalArrangement = Arrangement.spacedBy(4.dp)
    ) {
        CalendarView.entries.forEach { view ->
            val selected = currentView == view
            Box(
                modifier = Modifier
                    .weight(1f)
                    .clip(RoundedCornerShape(9.dp))
                    .background(if (selected) colors.primary else Color.Transparent)
                    .clickable { onViewChange(view) }
                    .padding(vertical = 9.dp),
                contentAlignment = Alignment.Center
            ) {
                Text(
                    text = if (view == CalendarView.GREGORIAN) "Gregorian" else "Igbo Era",
                    fontSize = 13.sp,
                    fontWeight = if (selected) FontWeight.Bold else FontWeight.Medium,
                    color = if (selected) colors.onPrimary else colors.onSurfaceMuted
                )
            }
        }
    }
}

@Composable
private fun GregorianPanel(
    state: CalendarUiState,
    onDateSelected: (LocalDate) -> Unit,
    onPrevMonth: () -> Unit,
    onNextMonth: () -> Unit
) {
    val monthFmt = DateTimeFormatter.ofPattern("MMMM yyyy")

    Column(modifier = Modifier.padding(horizontal = 16.dp)) {
        // Month navigation
        MonthNavRow(
            title = state.gregorianDisplayMonth.format(monthFmt),
            subtitle = IgboCalendarEngine.toIgboDate(state.gregorianDisplayMonth).run {
                if (isUboshiOmumu) "Ubọchị Ọmụmụ" else "$month · $year"
            },
            onPrev = onPrevMonth,
            onNext = onNextMonth
        )

        Spacer(Modifier.height(12.dp))

        // IZU legend bar
        IzuLegendBar()

        Spacer(Modifier.height(10.dp))

        // Calendar grid
        GregorianCalendarGrid(
            displayMonth = state.gregorianDisplayMonth,
            selectedDate = state.selectedDate,
            onDateSelected = onDateSelected
        )
    }
}

@Composable
private fun IgboEraPanel(
    state: CalendarUiState,
    onDateSelected: (LocalDate) -> Unit,
    onPrevMonth: () -> Unit,
    onNextMonth: () -> Unit
) {
    val colors = igboColors
    val anchorIgbo = IgboCalendarEngine.toIgboDate(state.igboDisplayAnchor)

    Column(modifier = Modifier.padding(horizontal = 16.dp)) {
        MonthNavRow(
            title = anchorIgbo.month,
            subtitle = "Aro ${anchorIgbo.year}  ·  Igbo Era",
            onPrev = onPrevMonth,
            onNext = onNextMonth
        )

        Spacer(Modifier.height(12.dp))

        if (!anchorIgbo.isUboshiOmumu) {
            // 4-day week structure info
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .clip(RoundedCornerShape(8.dp))
                    .background(colors.surfaceVariant)
                    .padding(horizontal = 12.dp, vertical = 8.dp),
                horizontalArrangement = Arrangement.SpaceBetween
            ) {
                Text(
                    "IZU (4-day week)",
                    fontSize = 11.sp,
                    color = colors.onSurfaceMuted,
                    fontWeight = FontWeight.SemiBold
                )
                Text(
                    "7 Izu = 28 days / Ọnwa",
                    fontSize = 11.sp,
                    color = colors.onSurfaceMuted
                )
            }

            Spacer(Modifier.height(10.dp))
        }

        IgboMonthGrid(
            anchorDate = state.igboDisplayAnchor,
            selectedDate = state.selectedDate,
            onDateSelected = onDateSelected
        )
    }
}

@Composable
private fun MonthNavRow(
    title: String,
    subtitle: String,
    onPrev: () -> Unit,
    onNext: () -> Unit
) {
    val colors = igboColors

    Row(
        modifier = Modifier.fillMaxWidth(),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.SpaceBetween
    ) {
        IconButton(onClick = onPrev) {
            Icon(
                Icons.Rounded.ChevronLeft,
                contentDescription = "Previous month",
                tint = colors.primary
            )
        }

        Column(horizontalAlignment = Alignment.CenterHorizontally) {
            Text(
                text = title,
                fontSize = 18.sp,
                fontWeight = FontWeight.Bold,
                color = colors.onBackground
            )
            Text(
                text = subtitle,
                fontSize = 11.sp,
                color = colors.onSurfaceMuted
            )
        }

        IconButton(onClick = onNext) {
            Icon(
                Icons.Rounded.ChevronRight,
                contentDescription = "Next month",
                tint = colors.primary
            )
        }
    }
}

@Composable
private fun IzuLegendBar() {
    val colors = igboColors
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(8.dp))
            .background(colors.surfaceVariant)
            .padding(horizontal = 8.dp, vertical = 6.dp),
        horizontalArrangement = Arrangement.SpaceEvenly,
        verticalAlignment = Alignment.CenterVertically
    ) {
        Text(
            "IZU (4-Day Week):",
            fontSize = 10.sp,
            color = colors.onSurfaceMuted,
            fontWeight = FontWeight.SemiBold
        )
        IgboCalendarEngine.IGBO_DAYS.forEachIndexed { idx, name ->
            Row(
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(3.dp)
            ) {
                Box(
                    Modifier
                        .size(6.dp)
                        .clip(androidx.compose.foundation.shape.CircleShape)
                        .background(com.igbocalendar.ui.theme.dayColor(idx))
                )
                Text(
                    name,
                    fontSize = 10.sp,
                    color = com.igbocalendar.ui.theme.dayColor(idx),
                    fontWeight = FontWeight.SemiBold
                )
            }
        }
    }
}

@Composable
private fun IzuInfoBanner(state: CalendarUiState) {
    val colors = igboColors
    val igbo = state.igboDate
    val izuDesc = com.igbocalendar.data.IzuData.weeks.getOrNull(igbo.weekIndex)

    if (izuDesc != null) {
        Column(
            modifier = Modifier
                .padding(horizontal = 16.dp)
                .clip(RoundedCornerShape(16.dp))
                .background(colors.surface)
                .border(1.dp, colors.divider, RoundedCornerShape(16.dp))
                .padding(16.dp)
        ) {
            Row(verticalAlignment = Alignment.CenterVertically) {
                Text(izuDesc.icon, fontSize = 18.sp)
                Spacer(Modifier.width(8.dp))
                Text(
                    text = izuDesc.name,
                    fontSize = 15.sp,
                    fontWeight = FontWeight.Bold,
                    color = colors.onSurface
                )
                Spacer(Modifier.weight(1f))
                Text(
                    text = izuDesc.tagline,
                    fontSize = 11.sp,
                    color = colors.primary,
                    fontWeight = FontWeight.Medium
                )
            }
            Spacer(Modifier.height(6.dp))
            Text(
                text = izuDesc.description,
                fontSize = 12.sp,
                color = colors.onSurfaceMuted,
                lineHeight = 18.sp
            )
        }
    }
}

File: ios/IgboCalendar/Screens/CalendarScreen.swift
import SwiftUI

// MARK: - Calendar Screen

struct CalendarScreen: View {
    @EnvironmentObject private var viewModel: CalendarViewModel
    @State private var showDateSearch = false

    private var state: CalendarUiState { viewModel.state }
    @Environment(\.igboColors) private var colors

    var body: some View {
        ScrollView {
            VStack(spacing: 0) {
                AppHeader(
                    onToggleTheme:  { viewModel.toggleTheme() },
                    onGoToday:      { viewModel.goToToday() },
                    onGoToDate:     { showDateSearch = true }
                )

                ViewToggleBar(
                    currentView:   state.calendarView,
                    onViewChange:  { viewModel.setCalendarView($0) }
                )

                Spacer().frame(height: 8)

                // Animated panel swap
                Group {
                    if state.calendarView == .gregorian {
                        GregorianPanel(
                            state:       state,
                            onDateSelected: { viewModel.selectDate($0) },
                            onPrevMonth:    { viewModel.gregorianPrevMonth() },
                            onNextMonth:    { viewModel.gregorianNextMonth() }
                        )
                        .transition(.opacity)
                    } else {
                        IgboEraPanel(
                            state:       state,
                            onDateSelected: { viewModel.selectDate($0) },
                            onPrevMonth:    { viewModel.igboPrevMonth() },
                            onNextMonth:    { viewModel.igboNextMonth() }
                        )
                        .transition(.opacity)
                    }
                }
                .animation(.easeInOut(duration: 0.2), value: state.calendarView)

                Spacer().frame(height: 16)

                IgboDateCard(
                    selectedDate: state.selectedDate,
                    igboDate:     state.igboDate
                )
                .padding(.horizontal, 16)

                if !state.igboDate.isUboshiOmumu {
                    Spacer().frame(height: 12)
                    IzuInfoBanner(igboDate: state.igboDate)
                        .padding(.horizontal, 16)
                }

                Spacer().frame(height: 24)
            }
        }
        .background(colors.background.ignoresSafeArea())
        .sheet(isPresented: $showDateSearch) {
            DateSearchSheet(
                currentDate:     state.selectedDate,
                currentIgboDate: state.igboDate,
                onNavigate:      { viewModel.selectDate($0) },
                onDismiss:       { showDateSearch = false }
            )
            .igboTheme(colors)
            .presentationDetents([.medium, .large])
            .presentationDragIndicator(.visible)
        }
    }
}

// MARK: - App Header

private struct AppHeader: View {
    let onToggleTheme: () -> Void
    let onGoToday:     () -> Void
    let onGoToDate:    () -> Void

    @Environment(\.igboColors) private var colors

    var body: some View {
        HStack {
            VStack(alignment: .leading, spacing: 2) {
                Text("Igbo Calendar")
                    .font(.system(size: 22, weight: .black))
                    .foregroundColor(colors.primary)
                Text("Ọha Igbo")
                    .font(.system(size: 12, weight: .medium))
                    .foregroundColor(colors.onSurfaceMuted)
            }

            Spacer()

            HStack(spacing: 8) {
                // Search button
                CircleButton(icon: "magnifyingglass", action: onGoToDate)

                // Today button
                Button(action: onGoToday) {
                    Text("Today")
                        .font(.system(size: 13, weight: .semibold))
                        .foregroundColor(colors.primary)
                        .padding(.horizontal, 14)
                        .padding(.vertical, 7)
                        .background(
                            RoundedRectangle(cornerRadius: 20)
                                .fill(colors.primaryContainer)
                        )
                }

                // Theme toggle
                Button(action: onToggleTheme) {
                    Text(colors.appTheme == .amber ? "🌙" : "🌕")
                        .font(.system(size: 16))
                        .frame(width: 36, height: 36)
                        .background(Circle().fill(colors.surfaceVariant))
                }
            }
        }
        .padding(.horizontal, 20)
        .padding(.vertical, 16)
    }
}

private struct CircleButton: View {
    let icon:   String
    let action: () -> Void

    @Environment(\.igboColors) private var colors

    var body: some View {
        Button(action: action) {
            Image(systemName: icon)
                .font(.system(size: 14, weight: .medium))
                .foregroundColor(colors.primary)
                .frame(width: 36, height: 36)
                .background(Circle().fill(colors.surfaceVariant))
        }
    }
}

// MARK: - View Toggle Bar

private struct ViewToggleBar: View {
    let currentView:  CalendarViewMode
    let onViewChange: (CalendarViewMode) -> Void

    @Environment(\.igboColors) private var colors

    var body: some View {
        HStack(spacing: 4) {
            ForEach([CalendarViewMode.gregorian, .igboEra], id: \.self) { view in
                let selected = currentView == view
                let label    = view == .gregorian ? "Gregorian" : "Igbo Era"
                Text(label)
                    .font(.system(size: 13, weight: selected ? .bold : .medium))
                    .foregroundColor(selected ? colors.onPrimary : colors.onSurfaceMuted)
                    .frame(maxWidth: .infinity)
                    .padding(.vertical, 9)
                    .background(
                        RoundedRectangle(cornerRadius: 9)
                            .fill(selected ? colors.primary : Color.clear)
                    )
                    .onTapGesture { onViewChange(view) }
            }
        }
        .padding(4)
        .background(RoundedRectangle(cornerRadius: 12).fill(colors.surfaceVariant))
        .padding(.horizontal, 20)
    }
}

// MARK: - Gregorian Panel

private struct GregorianPanel: View {
    let state:          CalendarUiState
    let onDateSelected: (Date) -> Void
    let onPrevMonth:    () -> Void
    let onNextMonth:    () -> Void

    @Environment(\.igboColors) private var colors

    private var monthTitle: String {
        let f = DateFormatter()
        f.dateFormat = "MMMM yyyy"
        return f.string(from: state.gregorianDisplayMonth)
    }

    private var igboSubtitle: String {
        let igbo = IgboCalendarEngine.toIgboDate(state.gregorianDisplayMonth)
        return igbo.isUboshiOmumu ? "Ubọchị Ọmụmụ" : "\(igbo.month) · \(igbo.year)"
    }

    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            MonthNavRow(
                title:    monthTitle,
                subtitle: igboSubtitle,
                onPrev:   onPrevMonth,
                onNext:   onNextMonth
            )

            Spacer().frame(height: 12)
            IzuLegendBar()
            Spacer().frame(height: 10)

            GregorianCalendarGrid(
                displayMonth:   state.gregorianDisplayMonth,
                selectedDate:   state.selectedDate,
                onDateSelected: onDateSelected
            )
        }
        .padding(.horizontal, 16)
    }
}

// MARK: - Igbo Era Panel

private struct IgboEraPanel: View {
    let state:          CalendarUiState
    let onDateSelected: (Date) -> Void
    let onPrevMonth:    () -> Void
    let onNextMonth:    () -> Void

    @Environment(\.igboColors) private var colors

    private var anchorIgbo: IgboDate {
        IgboCalendarEngine.toIgboDate(state.igboDisplayAnchor)
    }

    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            MonthNavRow(
                title:    anchorIgbo.month,
                subtitle: "Aro \(anchorIgbo.year)  ·  Igbo Era",
                onPrev:   onPrevMonth,
                onNext:   onNextMonth
            )

            Spacer().frame(height: 12)

            if !anchorIgbo.isUboshiOmumu {
                HStack {
                    Text("IZU (4-day week)")
                        .font(.system(size: 11, weight: .semibold))
                        .foregroundColor(colors.onSurfaceMuted)
                    Spacer()
                    Text("7 Izu = 28 days / Ọnwa")
                        .font(.system(size: 11))
                        .foregroundColor(colors.onSurfaceMuted)
                }
                .padding(.horizontal, 12)
                .padding(.vertical, 8)
                .background(RoundedRectangle(cornerRadius: 8).fill(colors.surfaceVariant))

                Spacer().frame(height: 10)
            }

            IgboMonthGrid(
                anchorDate:     state.igboDisplayAnchor,
                selectedDate:   state.selectedDate,
                onDateSelected: onDateSelected
            )
        }
        .padding(.horizontal, 16)
    }
}

// MARK: - Month Navigation Row

struct MonthNavRow: View {
    let title:    String
    let subtitle: String
    let onPrev:   () -> Void
    let onNext:   () -> Void

    @Environment(\.igboColors) private var colors

    var body: some View {
        HStack {
            Button(action: onPrev) {
                Image(systemName: "chevron.left")
                    .font(.system(size: 18, weight: .semibold))
                    .foregroundColor(colors.primary)
                    .frame(width: 40, height: 40)
            }

            Spacer()

            VStack(spacing: 2) {
                Text(title)
                    .font(.system(size: 18, weight: .bold))
                    .foregroundColor(colors.onBackground)
                Text(subtitle)
                    .font(.system(size: 11))
                    .foregroundColor(colors.onSurfaceMuted)
            }

            Spacer()

            Button(action: onNext) {
                Image(systemName: "chevron.right")
                    .font(.system(size: 18, weight: .semibold))
                    .foregroundColor(colors.primary)
                    .frame(width: 40, height: 40)
            }
        }
    }
}

// MARK: - IZU Legend Bar

private struct IzuLegendBar: View {
    @Environment(\.igboColors) private var colors

    var body: some View {
        HStack(spacing: 8) {
            Text("IZU (4-Day Week):")
                .font(.system(size: 10, weight: .semibold))
                .foregroundColor(colors.onSurfaceMuted)

            ForEach(Array(IgboCalendarEngine.igboDays.enumerated()), id: \.offset) { idx, name in
                HStack(spacing: 3) {
                    Circle()
                        .fill(Color.dayColor(idx))
                        .frame(width: 6, height: 6)
                    Text(name)
                        .font(.system(size: 10, weight: .semibold))
                        .foregroundColor(Color.dayColor(idx))
                }
            }
        }
        .padding(.horizontal, 8)
        .padding(.vertical, 6)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(RoundedRectangle(cornerRadius: 8).fill(colors.surfaceVariant))
    }
}

// MARK: - Izu Info Banner

private struct IzuInfoBanner: View {
    let igboDate: IgboDate

    @Environment(\.igboColors) private var colors

    private var izuDesc: IzuInfo? {
        guard igboDate.weekIndex >= 0 else { return nil }
        return IzuData.weeks.first { $0.index == igboDate.weekIndex }
    }

    var body: some View {
        if let izu = izuDesc {
            VStack(alignment: .leading, spacing: 6) {
                HStack {
                    Text(izu.icon).font(.system(size: 18))
                    Text(izu.name)
                        .font(.system(size: 15, weight: .bold))
                        .foregroundColor(colors.onSurface)
                    Spacer()
                    Text(izu.tagline)
                        .font(.system(size: 11, weight: .medium))
                        .foregroundColor(colors.primary)
                }
                Text(izu.description)
                    .font(.system(size: 12))
                    .foregroundColor(colors.onSurfaceMuted)
                    .lineSpacing(4)
            }
            .padding(16)
            .background(
                RoundedRectangle(cornerRadius: 16)
                    .fill(colors.surface)
                    .overlay(
                        RoundedRectangle(cornerRadius: 16)
                            .strokeBorder(colors.divider, lineWidth: 1)
                    )
            )
        }
    }
}

// MARK: - CalendarViewMode Hashable

extension CalendarViewMode: Hashable {}
